2-mandatory/3-shopping-list.js

```diff
@@ -19,8 +19,17 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
-}
+let missingIngredients = recipe.ingredients.filter(ingredient => {
+     return !pantry.fridgeContents.includes(ingredient) &&
+      !pantry.cupboardContents.includes(ingredient);
+  });
+ 
```

Comment:
great job :) you can also use concat like this:  `pantry.fridgeContents.concat(pantry.cupboardContents) and just check if you are missing something in this array or not`